### ISSUE TITLE
ERR_BLOCKED_BY_RESPONSE

### ISSUE THREAD

**Issue by lotigara** [Thesola10/nextcloud-nextframe#4]
Hello! I have running Nextcloud 29 at http://nextcloud/ and Forgejo (self-hosted GitHub) at http://forgejo/. When I'm trying to use the code from the [README.md](https://github.com/Thesola10/nextcloud-nextframe/blob/master/README.md) (I also use iFrame Resizer) on my Forgejo instance (Forgejo supports custom HTML, CSS and JS), Chromium shows the ERR_BLOCKED_BY_RESPONSE error.

P.S. I whitelisted http://forgejo/ in the Additional Settings, but that does nothing.

**Comment by Thesola10** [comment]
You may need to configure your Nextcloud to disable `X-Frame-Options` or hide it in a proxy such as Nginx or Apache

See [my config](https://cloud.thesola.io/git/nixos/serveur-maison/src/commit/c6bca472db5d004a38d84a28ed6a28d253a97b59/server/nginx.nix#L58)

**Comment by lotigara** [comment]
> See [my config](https://cloud.thesola.io/git/nixos/serveur-maison/src/commit/c6bca472db5d004a38d84a28ed6a28d253a97b59/server/nginx.nix#L58)

Sorry for late responce. How does this read in the nginx.conf format?

**Comment by Thesola10** [comment]
> > See [my config](https://cloud.thesola.io/git/nixos/serveur-maison/src/commit/c6bca472db5d004a38d84a28ed6a28d253a97b59/server/nginx.nix#L58)
> 
> Sorry for late responce. How does this read in the nginx.conf format?

Oh, the `extraConfig` string is literally Nginx config lines, just add the `proxy_hide_header` line to your config

**Comment by lotigara** [comment]
Thank you! I used the `Content-Security-Policy` HTTP header, because Chromium supports only that header, not the `X-Frame-Options`.

**Comment by lotigara** [comment]
I'm sorry for taking your time

**Comment by Thesola10** [comment]
> Thank you! I used the `Content-Security-Policy` HTTP header, because Chromium supports only that header, not the `X-Frame-Options`.

No, don't modify `Content-Security-Policy`, it should be managed by Nextframe directly.

Can you run `curl -i https://Nextcloud/apps/nextframe/disp/(token)` to see if it is passed through?

**Comment by lotigara** [comment]
> Can you run `curl -i https://Nextcloud/apps/nextframe/disp/(token)` to see if it is passed through?

Yes, it was passed through:
```
# curl -i https://Nextcloud/apps/nextframe/disp/(token)
...
content-security-policy: upgrade-insecure-requests
...
```

**Comment by Thesola10** [comment]
That doesn't look right. The URLs you added to "Allowed Ancestors" under "Additional Settings" should show up in the header.

Your proxy must be messing with it.

In fact, it's even missing the CSP entries that Nextcloud would usually add, even without Nextframe.

**Comment by Thesola10** [comment]
Here is what it _should_ look like:

```
content-security-policy: default-src 'none'; ...(default nextcloud stuff)...; frame-ancestors 'self' (extra ancestors);worker-src blob: 'self';form-action 'self'
```

**Comment by lotigara** [comment]
It is interesting... What nginx configs does NextFrame require for embedded page header to work? I have nothing in YunoHost's (nor official) Nextcloud nginx config associated with the `Content-Security-Policy` HTTP header.
